drivers/pn532/pn532.c

```diff
@@ -213,17 +214,16 @@ static int _read(pn532_t *dev, char *buff, unsigned len)
     }
     else {
 #ifdef PN532_SUPPORT_SPI
-        spi_acquire(dev->conf->spi);
+        spi_acquire(dev->conf->spi, GPIO_UNDEF, SPI_MODE, SPI_CLK);
         gpio_clear(dev->conf->nss);
-        spi_transfer_byte(dev->conf->spi, SPI_DATA_READ, NULL);
-        ret = spi_transfer_bytes(dev->conf->spi, 0x00, &buff[1], len);
+        spi_transfer_byte(dev->conf->spi, GPIO_UNDEF, true, SPI_DATA_READ);
+        spi_transfer_bytes(dev->conf->spi, GPIO_UNDEF, true, NULL, &buff[1], len);
         gpio_set(dev->conf->nss);
         spi_release(dev->conf->spi);
-        if (ret >= 0) {
-            buff[0] = 0x80;
-            reverse(buff, ret);
-            ret += 1;
-        }
+
+        buff[0] = 0x80;
+        reverse(buff, ret);
```

Comment:
no idea what happened, but this fix didn't survive the squash operation (: please replace `ret` with `len`.